src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx

```diff
@@ -85,62 +135,95 @@ function BulkEmailForm(props) {
   };
 
   const createEmailTask = async () => {
-    const emailData = new FormData();
     if (validateEmailForm()) {
       setEmailFormStatus(() => FORM_SUBMIT_STATES.PENDING);
-      emailData.append('action', 'send');
-      emailData.append('send_to', JSON.stringify(selectedRecipients));
-      emailData.append('subject', editor.emailSubject);
-      emailData.append('message', editor.emailBody);
-      if (isScheduled) {
-        emailData.append('schedule', new Date(`${editor.scheduleDate} ${editor.scheduleTime}`).toISOString());
-      }
-      let data;
-      try {
-        data = await postBulkEmail(emailData, courseId);
-      } catch (e) {
-        setEmailFormStatus(FORM_SUBMIT_STATES.ERROR);
-        return;
+      let response;
+      if (editor.editMode) {
+        const editedEmail = formatDataForFormAction(FORM_ACTIONS.PATCH);
+        response = await dispatch(editScheduledEmailThunk(editedEmail, courseId, editor.schedulingId));
+      } else {
+        const emailData = formatDataForFormAction(FORM_ACTIONS.POST);
+        response = await dispatch(postBulkEmailThunk(emailData, courseId));
       }
-      if (data.status === 200) {
+      if (response?.status === 200) {
```

Comment:
For my own edification, what does `response?` mean here? Or, more specifically, what does the `?` do here?

Reply:
A fairly new Js thing called [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). its the same as `response && response.status === 200`.

Reply:
Oh cool! Thanks for the link. :)